tabular/src/autogluon/tabular/models/abstract/model_trial.py

```diff
@@ -60,15 +74,26 @@ def fit_and_save_model(model, params, fit_args, predict_proba_args, y_val, time_
     else:
         time_left = None
 
+    if epochs is None:
+        epochs = 1
+
     model.params.update(params)
     time_fit_start = time.time()
-    model.fit(**fit_args, time_limit=time_left, reporter=reporter)
-    time_fit_end = time.time()
-    y_pred_proba = model.predict_proba(**predict_proba_args)
-    time_pred_end = time.time()
-    model.val_score = model.score_with_y_pred_proba(y=y_val, y_pred_proba=y_pred_proba)
-    model.fit_time = time_fit_end - time_fit_start
-    model.predict_time = time_pred_end - time_fit_end
+    incremental = epochs > 1
+    for i in range(epochs):
```

Comment:
if a model has num_iterations = 10 and epochs=5, it will train a total of 50 iterations (10 per epoch). Shouldn't it instead train a total of 10 iterations (2 per epoch)?

Reply:
My current design is num_iter per epoch actually. I  see this may be confusing. My concern is that if user specify a num_iter that's not divisible by epochs, should I do floor(iteration / epochs)?

Reply:
You should do a division such that each epoch can train a different amount that in total sum to the value:

X iter, Y epoch: Iterations per epoch

Ex: 5 iter, 3 epoch: [2, 2, 1]

Ex: 5 iter 6 epoch: [1, 1, 1, 1, 1, 0] (last epoch is skipped as if early stopping triggered)